File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.19-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_19() {
    let chars = ['a', 'b', 'c', 'd', 'e'];

    let row = 5;
    let column = 10;

    for (let y = row - 1; y >= 0; y--) {
        for (let x = column; x >= 0; x--) {
            if (x <= column / 2) {
                if (x > y) {
                    log(chars[x - 1]);
                } else {
                    log(' ');
                }
            } else if (x - 1 > column / 2) {
                let element = column - x;

                if (element >= y) {
                    log(chars[element]);
                } else {
                    log(' ');
                }
            }
        }

        console.log(y);
    }
}

pattern_19();

File: python/00-Basics/00.03-build-up-logical-thinking/0.3.19-pattern.py
from utils.clear import clear

def pattern_19():
    row = 5
    column = 10

    chars = ['a', 'b', 'c', 'd', 'e']

    for i in range(row, 0, -1):
        for j in range(1, column ):
            if j <= column / 2:
                if j >= i:
                    print(chars[j-1], end="")
                elif j < i:
                    print(" ", end="")

            elif j > column / 2:
                key = column - j
                if key >= i:
                    print(chars[key-1], end="")
                elif key < i:
                    print(" ", end="")

        print()



clear()
pattern_19()
